packages/host/src/lib/cpm-connector.ts

```diff
@@ -419,11 +407,14 @@ export class CPMConnector extends TypedEmitter<Events> {
         if (shouldReconnect) {
             this.isReconnecting = true;
 
-            setTimeout(async () => {
-                this.logger.info("Connection lost, retrying", this.connectionAttempts);
+            await new Promise<void>((res) => {
+                setTimeout(async () => {
+                    this.logger.info("Connection lost, retrying", this.connectionAttempts);
 
-                await this.connect();
-            }, this.RECONNECT_INTERVAL);
+                    await this.connect();
```

Comment:
Promise rejection should be handlerd - I'd just do:

```suggestion
                    this.connect().then(resolve, reject);
```

A rejection in timeout would reject the upper promise.